### ISSUE TITLE
20 second delays when running in EKS with 0.5.0

### ISSUE THREAD

**Issue by jasonmcintosh** [kubernetes-sigs/aws-iam-authenticator#347]
In testing upgrades of our docker images in EKS to use 0.5.0, we're seeing `aws-iam-authenticator token -i <clustername>` calls take 20 seconds before returning a token consistently.  It appears to be trying to hit the metadata endpoint repeatedly, then finally failing and using a fallback of some sort and immediately working.  Going back to 0.4.0 this problem goes away.  This does NOT happen when running on an EC2 node with 0.5.0.  Looking for any hints on how to debug this, or help?  Doesn't seem to be a configuration issue that I can tell.  We're using a pretty standard EKS deployment (though clusters are on mixes of kubernetes versions, most are at 15).

**Comment by micahhausler** [comment]
This is most likely due to the changes around IMDSv2 in the AWS SDK (ex: aws/aws-sdk-go#2972). Can you update to v0.5.2?

**Comment by jasonmcintosh** [comment]
Some test results:
0.5.2
```
{"Version":"0.5.2","Commit":"292b9b82df69b87af962b92485b254d9f4b10f00"}
bash-5.0$ time ./aws-iam-authenticator token -i test
real	0m3.148s
```

0.5.0
```
{"Version":"0.5.0","Commit":"1cfe2a90f68381eacd7b6dcfa2bf689e76eb8b4b"}
time ./aws-iam-authenticator token -i test
real	0m20.400s
```

0.4.0
```
{"Version":"0.4.0","Commit":"9999cd8047782c94c08e2d24b77dc9488947aa03"}
time ./aws-iam-authenticator token -i test
real	0m0.051s
```

SO 0.5.2 even adds 3 seconds to every call from testing.  0.4.0 is incredibly fast.  0.5.2 is better than 20 seconds, but still not what I'd consider "great"

**Comment by micahhausler** [comment]
How are you supplying credentials to your process?

**Comment by jasonmcintosh** [comment]
We spin up using the https://registry.terraform.io/modules/terraform-aws-modules/eks/aws ec2 instances with an IAM profiles.  I've seen this behavior on clusters running 1.14, 1.15 and 1.16.  I don't believe any our clusters are using OIDC auth at this point, though we've started looking at the service account - IAM role mappings, the pods where I've tested this performance are not using them currently, just instance profiles.
